frescobaldi_app/quickinsert/spanners.py

```diff
@@ -296,6 +300,107 @@ def spanner_positions(cursor):
+    """ Return the correct positions of the melisma commands, after all spanners already attached to the note """
+
+    def search_spanners_symbols(item1, item2, spanners) -> int:
+        """Find the spanners in the list lying between two music items.
+        :param item1: First item
+        :param item2: Second item
+        :param spanners: A list of spanners to be examined
+        :return: The position in the document of the last spanner attached to the first item, or None if nothing is found
+        """
+
+        last = None
+        for span in spanners:
+            if span[1] < item2.pos and span[0] >= item1.end:
+                last = span[1]
+        return last
+
+    crs = lydocument.cursor(cursor)
+    if cursor.hasSelection():
+        partial = ly.document.INSIDE
+    else:
+        # just select until the end of the current line
+        crs.select_end_of_block()
+        partial = ly.document.OUTSIDE
+
+    items = list(ly.rhythm.music_items(crs, partial=partial))
+
+    # only one item at the end of a block
+    if len(items) == 1:
+        return []
+
+    if cursor.hasSelection():
+        # get the music item next to the selection end (used to search for spanner symbols after the selection)
+        tk = next_music_token(crs.document, items[-1].end)
+        items.append(tk or _item(crs.end))
+        crs.select_end_of_block()  # need for search all spanners outside the selection
+    else:
+        del items[3:]  # leave at most 3 items
+        if len(items) < 3:
+            items.append(_item(crs.end))
+
+    # get all spanners from start up to last item
+    positions = get_spanners_in_range(crs.document, items[0].end, items[-1].pos)
+
+    a = search_spanners_symbols(items[0], items[1], positions) or items[0].end  # spanners between item0 and item1
+    b = search_spanners_symbols(items[-2], items[-1], positions) or items[-2].end  # spanners between the last two items
+
+    positions = []
+    for i in [a, b]:
+        c = QTextCursor(cursor.document())
+        c.setPosition(i)
+        positions.append(c)
+    return positions
+
+
+def _item(pos: int, end=None):
+    """Shorthand to get an empty music-item carrying positional information"""
+    return ly.rhythm.music_item(None, None, None, None, pos, end or pos)
+
+
+def next_music_token(doc, pos):
+    """
+    Return a music_item with its positions in the document next to the index. It identifies only the
+    start of a music item (Note, Rest, Skip, Chord), without collect all subsequent tokens (durations,
+    closing chords...)
+    :param doc: a lilypond document
+    :param pos: int indicating a position in the document
+    :return: (token.pos, token.end)
+    """
+    cursor = ly.document.Cursor(doc, pos)
+    source = ly.document.Source(cursor, True, ly.document.INSIDE, True)
+
+    for token in source:
+        if token.pos >= pos:
+            if isinstance(token, Note) or isinstance(token, Rest) \
```

Comment:
suggestion: `isinstance` takes a tuple of types as the second argument as in this borrowed example `isinstance("Hello", (float, int, str, list, dict, tuple))` so you don't need to do a lot of "ors".